api/src/main/java/com/example/api/service/user/UserService.java

```diff
@@ -147,4 +157,52 @@ public String getProfessorRegisterToken() throws WrongUserTypeException {
         log.info("Professor {} fetch ProfessorRegisterToken", user.getEmail());
         return professorRegisterToken.getToken();
     }
+
+    public void deleteProfessorAccount(String email) throws WrongUserTypeException {
+        User user = getCurrentUser();
+        User professor = userRepo.findUserByEmail(email);
```

Comment:
ja bym dał zamiast `professor` coś w stylu `newProfessor` żeby podkreślić że chodzi o podmianę osoby przypisanej. Tak samo z parametrem `email` - zmienił bym nazwę na taką, która wskazuje, że ten email to jest email nowego profesora dla aktywności